app/views/inventory/checklist.html.erb

```diff
@@ -5,43 +5,75 @@
   <small>Note: blank values will not be saved</small>
 </header>
 
-<table>
+<table id="inventory_table" class="tablesorter {sortlist: [[2,0]]}">
   <thead>
-    <th>Done?</th>
+    <th></th>
+    <th></th>
+    <th>Author</th>
     <th>Title</th>
     <th>ISBN</th>
-    <th>Publisher</th>
     <th>Publish Date</th>
-    <th>Publication Place</th>
-    <th>Language</th>
-    <th>Pages</th>
     <th>Location</th>
     <th>Count</th>
     <th>In Storage</th>
     <th>Missing</th>
     <th>Notable</th>
     <th>Keep Multiple</th>
+    <th>Done?</th>
   </thead>
   <tbody>
     <% @genre.books.each do |book| %>
-      <tr>
-        <td><input type="checkbox" onchange="sendData(<%= book.id %>)" class="inventory-commit"></td>
+      <tr class="<%= 'complete' if book.inventoried %>">
+        <td><span class="<%= 'unavailable' if !book.available %>" title="unavailable"></span></td>
+        <td><%=  link_to "edit", edit_book_path(book), target: "_blank" %></td>
+        <td><%= book.primary_author.name %></td>
         <td><input type="text" value="<%= book.title %>" class="book_<%= book.id %>" name="title"></td>
         <td><input type="text" value="<%= book.isbn %>" class="book_<%= book.id %>" name="isbn"></td>
-        <!-- <td><input type="text" value="<%= book.publisher %>" class="book_<%= book.id %>" name="publisher"></td> -->
         <td><input type="text" value="<%= book.publish_date %>" class="book_<%= book.id %>" name="publish_date"></td>
-        <td><input type="text" value="<%= book.publication_place %>" class="book_<%= book.id %>" name="publication_place"></td>
-        <td><input type="text" value="<%= book.language %>" class="book_<%= book.id %>" name="language"></td>
-        <td><input type="text" value="<%= book.pages %>" class="book_<%= book.id %>" name="pages"></td>
         <td><input type="text" value="<%= book.location %>" class="book_<%= book.id %>" name="location"></td>
         <td><input type="number" value="<%= book.count %>" class="book_<%= book.id %>" name="count"></td>
-        <td><input type="checkbox" value="<%= book.in_storage %>" class="book_<%= book.id %>" name="in_storage"></td>
-        <td><input type="checkbox" value="<%= book.missing %>" class="book_<%= book.id %>" name="missing"></td>
-        <td><input type="checkbox" value="<%= book.notable %>" class="book_<%= book.id %>" name="notable"></td>
-        <td><input type="checkbox" value="<%= book.keep_multiple %>" class="book_<%= book.id %>" name="keep_multiple"></td>
+        <td>
+          <input type="hidden" value="false" class="book_<%= book.id %>" name="in_storage">
+          <% if book.in_storage %>
+            <input type="checkbox" value="true" class="book_<%= book.id %>" name="in_storage" checked >
```

Comment:
Correct. It'll submit the original value for the field.

Afaik, the value argument is the value if the box is checked.
On Aug 23, 2015 6:08 PM, "cheri allen" <notifications@github.com> wrote:

> In app/views/inventory/checklist.html.erb
> <https://github.com/cherimarie/LibraryCollection/pull/66#discussion_r37713414>
> :
>
> >          <td><input type="text" value="<%= book.location %>" class="book_<%= book.id %>" name="location"></td>
> >          <td><input type="number" value="<%= book.count %>" class="book_<%= book.id %>" name="count"></td>
> > -        <td><input type="checkbox" value="<%= book.in_storage %>" class="book_<%= book.id %>" name="in_storage"></td>
> > -        <td><input type="checkbox" value="<%= book.missing %>" class="book_<%= book.id %>" name="missing"></td>
> > -        <td><input type="checkbox" value="<%= book.notable %>" class="book_<%= book.id %>" name="notable"></td>
> > -        <td><input type="checkbox" value="<%= book.keep_multiple %>" class="book_<%= book.id %>" name="keep_multiple"></td>
> > +        <td>
> > +          <input type="hidden" value="false" class="book_<%= book.id %>" name="in_storage">
> > +          <% if book.in_storage %>
> > +            <input type="checkbox" value="true" class="book_<%= book.id %>" name="in_storage" checked >
>
> So, doing value="<%=book.in_storage %>" doesn't work for this? Instead of
> hidden field?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cherimarie/LibraryCollection/pull/66/files#r37713414>.
>